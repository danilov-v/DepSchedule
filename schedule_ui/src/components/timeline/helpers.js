import { getDayWithoutMinutes } from "utils/date";

export const getLastGenUnits = units => {
  const getUnitLastChilds = unit =>
    unit.childUnit && unit.childUnit.length
      ? unit.childUnit.map(getUnitLastChilds)
      : [unit];

  return units
    .map(unit => getUnitLastChilds(unit))
    .reduce((lastChilds, group) => [...lastChilds, ...group], [])
    .flat(5);
};

export const formatPeriods = periods =>
  periods.map(period => ({
    ...period,
    startDate: getDayWithoutMinutes(new Date(period.startDate)),
    endDate: getDayWithoutMinutes(new Date(period.endDate)),
  }));
