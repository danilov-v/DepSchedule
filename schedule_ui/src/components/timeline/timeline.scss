.timeline {
  &-wrapper {
    overflow: auto;
    height: 600px;
    border: 1px solid black;
  }

  &-info {
    width: 200px;
  }

  &-left {
    max-width: max-content;
  }
}
