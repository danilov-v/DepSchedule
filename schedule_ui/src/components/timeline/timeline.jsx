import React, { useState } from "react";
import { Container, Row, Col } from "reactstrap";
import { addMonths, addDays } from "date-fns";
import { Title } from "components/title/title";
import { Calendar } from "components/calendar/calendar";
import { HighLevelSections } from "components/high-level-sections/high-level-sections";

import { getDates } from "utils/date";

import "./timeline.scss";

export function Timeline() {
  const now = new Date();

  const [startDate, setStartDate] = useState(now);
  const [endDate, setEndDate] = useState(addMonths(now, 3));

  const sections = [
    {
      name: "A",
      startDate: new Date(2019, 7, 16),
      endDate: new Date(2019, 8, 16),
    },
    {
      name: "B",
      startDate: new Date(2019, 8, 17),
      endDate: new Date(2019, 8, 26),
    },
    {
      name: "C",
      startDate: new Date(2019, 8, 27),
      endDate: new Date(2019, 9, 10),
    },
    {
      name: "D",
      startDate: new Date(2019, 9, 12),
      endDate: new Date(2019, 9, 16),
    },
  ];

  // const sections = [
  //   {
  //     name: "A",
  //     startDate: new Date(2019, 6, 16),
  //     endDate: new Date(2019, 8, 16),
  //   },
  // ];

  console.log(sections);

  const range = getDates(startDate, endDate);

  return (
    <Container className="timeline">
      <Title
        onChangeStartDate={setStartDate}
        onChangeEndDate={setEndDate}
        startDate={startDate}
        endDate={endDate}
      />
      <div className="timeline-wrapper">
        <Row>
          <Col>
            <HighLevelSections
              startDate={startDate}
              range={range}
              sections={sections}
            />
          </Col>
        </Row>
        <Row className="flex-nowrap" noGutters>
          <Col className="timeline-left" xs="auto">
            <Calendar range={range} />
          </Col>
          <Col xs="auto" className="timeline-info">
            <div>test</div>
          </Col>
        </Row>
      </div>
    </Container>
  );
}
