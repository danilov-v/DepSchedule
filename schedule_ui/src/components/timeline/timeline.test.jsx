import React from "react";
import { shallow } from "enzyme";

import { Timeline } from "./timeline";

describe("timeline.jsx", () => {
  describe("<Timeline />", () => {
    it("should render and match snapshot", () => {
      const component = shallow(<Timeline />);

      expect(component).toMatchSnapshot();
    });
  });
});
