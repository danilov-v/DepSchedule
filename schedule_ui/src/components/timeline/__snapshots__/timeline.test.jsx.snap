// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timeline.jsx <Timeline /> should render and match snapshot 1`] = `
<Container
  className="timeline"
  tag="div"
>
  <Title
    endDate={2019-10-19T08:00:19.678Z}
    onChangeEndDate={[Function]}
    onChangeStartDate={[Function]}
    startDate={2019-07-19T08:00:19.678Z}
  />
  <div
    className="timeline-wrapper"
  >
    <Row
      className="flex-nowrap"
      noGutters={true}
      tag="div"
    >
      <Col
        className="timeline-left"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="auto"
      >
        <Calendar
          range={
            Array [
              Object {
                "days": Array [
                  2019-07-19T08:00:19.678Z,
                  2019-07-20T08:00:19.678Z,
                  2019-07-21T08:00:19.678Z,
                  2019-07-22T08:00:19.678Z,
                  2019-07-23T08:00:19.678Z,
                  2019-07-24T08:00:19.678Z,
                  2019-07-25T08:00:19.678Z,
                  2019-07-26T08:00:19.678Z,
                  2019-07-27T08:00:19.678Z,
                  2019-07-28T08:00:19.678Z,
                  2019-07-29T08:00:19.678Z,
                  2019-07-30T08:00:19.678Z,
                  2019-07-31T08:00:19.678Z,
                ],
                "name": "July",
              },
              Object {
                "days": Array [
                  2019-08-01T08:00:19.678Z,
                  2019-08-02T08:00:19.678Z,
                  2019-08-03T08:00:19.678Z,
                  2019-08-04T08:00:19.678Z,
                  2019-08-05T08:00:19.678Z,
                  2019-08-06T08:00:19.678Z,
                  2019-08-07T08:00:19.678Z,
                  2019-08-08T08:00:19.678Z,
                  2019-08-09T08:00:19.678Z,
                  2019-08-10T08:00:19.678Z,
                  2019-08-11T08:00:19.678Z,
                  2019-08-12T08:00:19.678Z,
                  2019-08-13T08:00:19.678Z,
                  2019-08-14T08:00:19.678Z,
                  2019-08-15T08:00:19.678Z,
                  2019-08-16T08:00:19.678Z,
                  2019-08-17T08:00:19.678Z,
                  2019-08-18T08:00:19.678Z,
                  2019-08-19T08:00:19.678Z,
                  2019-08-20T08:00:19.678Z,
                  2019-08-21T08:00:19.678Z,
                  2019-08-22T08:00:19.678Z,
                  2019-08-23T08:00:19.678Z,
                  2019-08-24T08:00:19.678Z,
                  2019-08-25T08:00:19.678Z,
                  2019-08-26T08:00:19.678Z,
                  2019-08-27T08:00:19.678Z,
                  2019-08-28T08:00:19.678Z,
                  2019-08-29T08:00:19.678Z,
                  2019-08-30T08:00:19.678Z,
                  2019-08-31T08:00:19.678Z,
                ],
                "name": "August",
              },
              Object {
                "days": Array [
                  2019-09-01T08:00:19.678Z,
                  2019-09-02T08:00:19.678Z,
                  2019-09-03T08:00:19.678Z,
                  2019-09-04T08:00:19.678Z,
                  2019-09-05T08:00:19.678Z,
                  2019-09-06T08:00:19.678Z,
                  2019-09-07T08:00:19.678Z,
                  2019-09-08T08:00:19.678Z,
                  2019-09-09T08:00:19.678Z,
                  2019-09-10T08:00:19.678Z,
                  2019-09-11T08:00:19.678Z,
                  2019-09-12T08:00:19.678Z,
                  2019-09-13T08:00:19.678Z,
                  2019-09-14T08:00:19.678Z,
                  2019-09-15T08:00:19.678Z,
                  2019-09-16T08:00:19.678Z,
                  2019-09-17T08:00:19.678Z,
                  2019-09-18T08:00:19.678Z,
                  2019-09-19T08:00:19.678Z,
                  2019-09-20T08:00:19.678Z,
                  2019-09-21T08:00:19.678Z,
                  2019-09-22T08:00:19.678Z,
                  2019-09-23T08:00:19.678Z,
                  2019-09-24T08:00:19.678Z,
                  2019-09-25T08:00:19.678Z,
                  2019-09-26T08:00:19.678Z,
                  2019-09-27T08:00:19.678Z,
                  2019-09-28T08:00:19.678Z,
                  2019-09-29T08:00:19.678Z,
                  2019-09-30T08:00:19.678Z,
                ],
                "name": "September",
              },
              Object {
                "days": Array [
                  2019-10-01T08:00:19.678Z,
                  2019-10-02T08:00:19.678Z,
                  2019-10-03T08:00:19.678Z,
                  2019-10-04T08:00:19.678Z,
                  2019-10-05T08:00:19.678Z,
                  2019-10-06T08:00:19.678Z,
                  2019-10-07T08:00:19.678Z,
                  2019-10-08T08:00:19.678Z,
                  2019-10-09T08:00:19.678Z,
                  2019-10-10T08:00:19.678Z,
                  2019-10-11T08:00:19.678Z,
                  2019-10-12T08:00:19.678Z,
                  2019-10-13T08:00:19.678Z,
                  2019-10-14T08:00:19.678Z,
                  2019-10-15T08:00:19.678Z,
                  2019-10-16T08:00:19.678Z,
                  2019-10-17T08:00:19.678Z,
                  2019-10-18T08:00:19.678Z,
                  2019-10-19T08:00:19.678Z,
                ],
                "name": "October",
              },
            ]
          }
        />
      </Col>
      <Col
        className="timeline-info"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="auto"
      >
        <div>
          test
        </div>
      </Col>
    </Row>
  </div>
</Container>
`;
